Sources/StreamChat/Controllers/PollController/PollController.swift

```diff
@@ -0,0 +1,283 @@
+            filter: .and(
+                [.equal(.userId, to: client.currentUserId ?? ""), .equal(.pollId, to: pollId)]
+            )
+        )
+        pollsRepository = client.pollsRepository
+        
+        super.init()
+        
+        setupPollObserver()
+    }
+    
+    override public func synchronize(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        startObserversIfNeeded()
+
+        pollsRepository.queryPollVotes(query: ownVotesQuery) { result in
+            if let error = result.error {
+                self.state = .remoteDataFetchFailed(ClientError(with: error))
+            } else {
+                self.state = .remoteDataFetched
+            }
+            self.callback { completion?(result.error) }
+        }
+    }
+    
+    /// Casts a vote for a poll.
+    ///
+    /// - Parameters:
+    ///   - answerText: An optional text answer for the poll.
+    ///   - optionId: An optional identifier for the poll option.
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func castPollVote(
+        answerText: String?,
+        optionId: String?,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        pollsRepository.castPollVote(
+            messageId: messageId,
+            pollId: pollId,
+            answerText: answerText,
+            optionId: optionId,
+            currentUserId: client.currentUserId,
+            query: ownVotesQuery,
+            completion: completion
+        )
+    }
+
+    /// Removes a vote from a poll.
+    ///
+    /// - Parameters:
+    ///   - voteId: The identifier of the vote to be removed.
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func removePollVote(
+        voteId: String,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        pollsRepository.removePollVote(
+            messageId: messageId,
+            pollId: pollId,
+            voteId: voteId,
+            completion: completion
+        )
+    }
+
+    /// Closes the poll.
+    ///
+    /// - Parameters:
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func closePoll(completion: ((Error?) -> Void)? = nil) {
+        pollsRepository.closePoll(pollId: pollId, completion: completion)
+    }
+
+    /// Suggests a new option for the poll.
+    ///
+    /// - Parameters:
+    ///   - text: The text of the suggested poll option.
+    ///   - position: An optional position for the suggested option.
+    ///   - custom: An optional dictionary of custom data for the suggested option.
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func suggestPollOption(
+        text: String,
+        position: Int? = nil,
+        custom: [String: RawJSON]? = nil,
```

Comment:
We use `userData` quite a bit for cases like this. Would it make sense to unify the naming here?

Reply:
It's actually `extraData` that we use currently, but we should use `custom` to align with other platforms. Now, the question is - do we want to add the new stuff the correct way, or we keep consistency with the rest of the codebase. Not sure, wdyt?